File: recipesv2/src/v1/desktop.py
from .enumeration import Relationship, Repo, Scope
from .atomic import Unit
from box import Box

m=Box()

m.dotfile_manager=Box(
    # stow=Box(repo=Repo.AOR),
    # chemzmoi=Box(repo=Repo.AOR),
)

m.window_compositor=Box(
    # niri=Box(repo=Repo.AOR),
    # river=Box(repo=Repo.AOR),
    # mangowc=Box(repo=Repo.AOR),
    hyprland=Box(
        repo=Repo.AOR,
        single_user_config=[
            "~/.config/hypr/hyprland",
            "~/.config/hypr/hyprland.conf",
        ]
    )
)

m.session_manager.relationship=Relationship.ASSOCIATED
m.session_manager=Box(
    app2unit=Box(repo=Repo.AOR),
    uwsm=Box(
        repo=Repo.AOR,
        single_user_config=["~./config/uwsm"],
        supporters=Box(libnewt=Box(repo=Repo.AOR))
    )
)

m.display_manager.sddm=Box(repo=Repo.AOR, units=[Unit("sddm.service", Scope.MULTI_USER)])

m.screen_share.relationship=Relationship.ASSOCIATED
m.screen_share=Box(
    xdg_desktop_portal_gtk=Box(repo=Repo.AOR),
    xdg_desktop_portal_hyprland=Box(repo=Repo.AOR)
)

m.bar=Box(
    waybar=Box(
        repo=Repo.AOR,
        units=[Unit("waybar.service", Scope.SINGLE_USER)],
        single_user_config=["~/.config/waybar"]
    ),
    # eww=Box(repo=Repo.AOR),
    # ags=Box(repo=Repo.AOR),
    # hyprpanel=Box(repo=Repo.AOR),
    # quickshell=Box(repo=Repo.AOR),
)

m.color_extractor=Box(
    # pywall=Box(repo=Repo.AOR),
    # wallust=Box(repo=Repo.AOR),
    # hellwal=Box(repo=Repo.AOR),
    matugen=Box(repo=Repo.AOR, single_user_config=["~/.config/matugen"]),
    # python_colorthief=Box(repo=Repo.AOR),
    # kde_material_you_colors=Box(repo=Repo.AOR),
)
m.qt_theme.relationship=Relationship.ASSOCIATED
m.qt_theme=Box(
    qt5ct=Box(repo=Repo.AOR, single_user_config=["~/.config/qt5ct"]),
    qt6ct=Box(repo=Repo.AOR, single_user_config=["~/.config/qt6ct"]),
    hyprqt6engine=Box(repo=Repo.AOR),
    kvantum=Box(repo=Repo.AOR, single_user_config=["~/.config/Kvantum"]),
)

m.gtk_theme.relationship=Relationship.ASSOCIATED
m.gtk_theme=Box(
    gtk=Box(
        single_user_config=[
            "~/.config/gtk-3.0",
            "~/.config/gtk-4.0"
        ]
    ),
    nwg_look=Box(repo=Repo.AOR, single_user_config=["~/.config/nwg-look"]),
    orchis_theme=Box(
        repo=Repo.AOR,
        supporters=Box(
            vimix_cursors=Box(repo=Repo.AOR),
            tela_circle_icon_theme=Box(repo=Repo.AOR),
        ),
    ),
    gnome_themes_extra=Box(repo=Repo.AOR),
)

m.wallpaper.setter=Box(
    # swww=Box(repo=Repo.AOR),
    # swaybg=Box(repo=Repo.AOR),
    # awww_bin=Box(repo=Repo.AUR),
    # mpvpaper=Box(repo=Repo.AUR),
    hyprpaper=Box(
        repo=Repo.AOR,
        units=[Unit("hyprpaper.service", Scope.SINGLE_USER)],
        single_user_config=["~/.config/hypr/hyprpaper.conf"]
    )
)

m.wallpaper.picker=Box(
    rofi=Box(repo=Repo.AOR, single_user_config=["~/.config/rofi"]),
    # waypaper=Box(repo=Repo.AUR),
    # waytrogen_bin=Box(repo=Repo.AUR),
)

m.screen_lock.hyprlock=Box(repo=Repo.AOR, single_user_config=["~/.config/hypr/hyprlock.conf"])
m.polkit.hyprpolkitagent=Box(repo=Repo.AOR, units=[Unit("hyprpolkitagent.service", Scope.SINGLE_USER)])
m.idle.hypridle=Box(
    repo=Repo.AOR,
    units=[Unit("hypridle.service", Scope.SINGLE_USER)],
    single_user_config=["~/.config/hypr/hypridle.conf"]
)
m.backlight.hyprsunset=Box(
    repo=Repo.AOR,
    units=[Unit("hyprsunset.service", Scope.SINGLE_USER)],
    single_user_config=["~/.config/hypr/hyprsunset.conf"]
)

m.launcher=Box(
    fuzzel=Box(repo=Repo.AOR, single_user_config=["~/.config/fuzzel"]),
    hyprlauncher=Box(repo=Repo.AOR, single_user_config=["~/.config/hypr/hyprlauncher.conf"]),
)

m.notification=Box(
    # fnott=Box(repo=Repo.AOR),
    # dunst=Box(repo=Repo.AOR, single_user_config=["~/.config/dunst"]),
    # mako=Box(repo=Repo.AOR, units=[Unit("mako.service", Scope.SINGLE_USER)], single_user_config=["~/.config/mako"]),
    swaync=Box(repo=Repo.AOR, units=[Unit("swaync.service", Scope.SINGLE_USER)], single_user_config=["~/.config/swaync"]),
)

m.clipboard.copy_paste.wl_clipboard=Box(repo=Repo.AOR)
m.clipboard.history=Box(
    # copyq=Box(repo=Repo.AOR),
    cliphist=Box(repo=Repo.AOR),
    # nwg_clipman=Box(repo=Repo.AOR),
    # wl_clllip_persist=Box(repo=Repo.AOR),
)

m.input.method.fcitx5=Box(
    repo=Repo.AOR,
    single_user_config=[
        "~/.config/fcitx5/config"
        "~/.config/fcitx5/profile"
    ],
    supporters=Box(
        fcitx5_unikey=Box(repo=Repo.AOR),
        fcitx5_configtool=Box(repo=Repo.AOR)
    )
)
m.input.remapper=Box(
    kanata_bin=Box(repo=Repo.AUR, single_user_config=["~/.config/kanata"]),
    # wlr_which_key=Box(repo=Repo.AUR),
    # xremap_hypr_bin=Box(repo=Repo.AUR),
    # xremap_wlroots_bin=Box(repo=Repo.AUR),
)

m.font.noto=Box(
    noto_fonts=Box(repo=Repo.AOR),
    noto_fonts_cjk=Box(repo=Repo.AOR),
    noto_fonts_emoji=Box(repo=Repo.AOR),
)
m.font.ttf=Box(
    ttf_opensans=Box(repo=Repo.AOR),
    ttf_fira_code=Box(repo=Repo.AOR),
    ttf_cascadia_code_nerd=Box(repo=Repo.AOR),
    ttf_jetbrains_mono_nerd=Box(repo=Repo.AOR),
)
m.font.tex.tex_gyre_fonts=Box(repo=Repo.AOR),

m.shell=Box(
    zsh=Box(repo=Repo.AOR),
    fish=Box(repo=Repo.AOR),
    bash=Box(repo=Repo.AOR, single_user_config=["~/.bashrc", "~/.bash_profile"]),
    nushell=Box(
        repo=Repo.AOR,
        single_user_config=[
            # "~/.config/nushell",
            "~/.config/nushell/config.nu",
            "~/.config/nushell/env.nu",
        ]
    )
)
m.shell.history.atuin=Box(repo=Repo.AOR)
m.shell.prompt.starship=Box(repo=Repo.AOR, single_user_config=["~/.config/starship.toml"])

m.terminal.emulator=Box(
    foot=Box(
        repo=Repo.AOR,
        units=[Unit("foot-server.service", Scope.SINGLE_USER)],
        single_user_config=["~/.config/foot"]
    ),
    wezterm_git=Box(
        repo=Repo.AUR,
        single_user_config=["~/.config/wezterm"]
    ),
    kitty=Box(repo=Repo.AOR, single_user_config=["~/.config/kitty"]),
    aclacritty=Box(repo=Repo.AOR),
    ghostty=Box(repo=Repo.AOR, units=[Unit("app-com.mitchellh.ghostty.service", Scope.SINGLE_USER)]),
)

m.terminal.multiplexer=Box(
    tmux=Box(repo=Repo.AOR, single_user_config=["~/.tmux.conf"]),
    zellij=Box(repo=Repo.AOR),
)

m.file.manager=Box(
    thunar=Box(
        repo=Repo.AOR,
        supporters=Box(
            thunar_volman=Box(repo=Repo.AOR),
            thunar_archive_plugin=Box(repo=Repo.AOR),
            thunar_media_tags_plugin=Box(repo=Repo.AOR),
            # catfish=Box(repo=Repo.AOR),
            # plocate=Box(repo=Repo.AOR),
            # zeitgeist=Box(repo=Repo.AOR),
        )
    ),
    yazi=Box(
        repo=Repo.AOR,
        single_user_config=["~/.config/yazi"],
        supporters={
            "7zip": {"repo": Repo.AOR},
            "chafa": {"repo": Repo.AOR},
            "ffmpeg": {"repo": Repo.AOR},
            "jq": {"repo": Repo.AOR},
            "poppler": {"repo":Repo.AOR},
            "resvg": {"repo":Repo.AOR},
            "imagemagick": {
                "repo": Repo.AOR,
                "supporters": {"libjpeg-turbo": {"repo": Repo.AOR}}
            }
        }
    ),
    superfile=Box(repo=Repo.AOR)
)
m.file.misc=Box(
    xdg_user_dirs=Box(repo=Repo.AOR, units=[Unit("xdg-user-dirs-update.service", Scope.SINGLE_USER)]),
    czkawka_gui_bin=Box(repo=Repo.AUR),
)

m.file.thumnail.tumbler=Box(
    repo=Repo.AOR,
    units=[Unit("tumblerd.service", Scope.SINGLE_USER)],
    supporters=Box(
        ffmpegthumbnailer=Box(repo=Repo.AOR),
        freetype2=Box(repo=Repo.AOR),
        libgepub=Box(repo=Repo.AOR),
        libgsf=Box(Repo.AOR),
        libopenraw=Box(repo=Repo.AOR),
        poppler_glib=Box(repo=Repo.AOR),
        libarchive=Box(repo=Repo.AOR),
        # ueberzugpp=Box(repo=Repo.AOR),
    )
)

m.media.videp=Box(
    mpv=Box(repo=Repo.AOR, single_user_config=["~/.config/mpv"]),
    # vlc=Box(repo=Repo.AOR),
    # clapper=Box(repo=Repo.AOR),
)
m.media.image=Box(
    mpv=Box(repo=Repo.AOR, single_user_config=["~/.config/mpv"]),
    imv=Box(repo=Repo.AOR),
    feh=Box(repo=Repo.AOR),
    gthumb=Box(repo=Repo.AOR),
    swayimg=Box(repo=Repo.AOR),
)

m.screen.capture=Box(
    # grim=Box(repo=Repo.AOR),
    # flameshot=Box(Repo.AOR),
    hyprshot=Box(repo=Repo.AOR, supporters=Box(hyprpicker=Box(repo=Repo.AOR)))
)
m.screen.crop.slurp=Box(repo=Repo.AOR)
m.screen.annotate=Box(
    swappy=Box(repo=Repo.AOR),
    satty=Box(repo=Repo.AOR),
)

m.screen.record=Box(
    wl_screenrec=Box(repo=Repo.AUR),
    obs_studio=Box(repo=Repo.AOR),
    wf_recorder=Box(repo=Repo.AOR),
    gpu_screen_recorder=Box(repo=Repo.AOR, supporters=Box(gpu_screen_recorder_ui=Box(repo=Repo.AOR)))
)

m.browser.gui=Box(
    firefox=Box(repo=Repo.AOR, supporters=Box(speech_dispatcher=Box(repo=Repo.AOR))),
    brave_bin=Box(repo=Repo.AUR),
    zen_browser_bin=Box(repo=Repo.AUR),
    google_chrome_bin=Box(repo=Repo.AUR, single_user_config=["~/.config/chrome-flags.conf"]),
    helium_browser_bin=Box(repo=Repo.AUR, single_user_config=["~/.config/helium-flags.conf"]),
    thorium_browser_bin=Box(repo=Repo.AUR, single_user_config=["~/.config/thorium-flags.conf"]),
    microsoft_edge_stable_bin=Box(repo=Repo.AUR),
)
m.browser.keyboard_driven=Box(
    browsh=Box(repo=Repo.AUR),
    nyxt=Box(repo=Repo.AOR),
    lynx=Box(repo=Repo.AOR),
    luakit=Box(repo=Repo.AOR),
)

m.misc=Box(
    hugo=Box(repo=Repo.AOR),
    anki_bin=Box(repo=Repo.AUR),
    okular=Box(repo=Repo.AOR),
    mediawriter=Box(repo=Repo.AOR),
    cava=Box(repo=Repo.AOR, single_user_config=["~/.config/cava"]),
    electron=Box(repo=Repo.AOR, single_user_config=["~/.config/electron-flags.conf"]),
    #TODO: this recipes need extension too, how to write
    python_pywalfox=Box(repo=Repo.AUR)
)

File: recipes/src/package_1/desktop.lua
dofile("utils.lua")

local m={}


m.dotfile_manager={
    stow=aor,
    -- chemzmoi=aor,
}

m.window_compositor={
    -- niri=aor,
    -- river=aor,
    -- mangowc=aor,
    hyprland={
        Repo.AOR,
        single_user_config={
            "~/.config/hypr/hyprland",
            "~/.config/hypr/hyprland.conf",
            {"sh", "~/.config/hypr/hyprland/scripts/first_intel_gpu.sh"},
            {"sh", "~/.config/hypr/hyprland/scripts/first_amd_gpu.sh"},
        },
    }
}

m.session_manager.relationship=Relationship.ASSOCIATED
m.session_manager={
    app2unit=aor,
    uwsm={
        Repo.AOR,
        single_user_config={"~./config/uwsm"},
        supporters={libnewt=aor}
    }
}

m.display_manager.sddm={
    Repo.AOR,
    units={"sddm.service", Scope.MULTI_USER},
    supporters={
        qt6_svg=aor,
        qt6_wayland=aor,
        qt6_virtualkeyboard=aor,
        qt6_multimedia_ffmpeg=aor
    },
    multiple_user_setting={
        "/etc/sddm.conf",
        "/etc/sddm.conf.d/virtualkbd.conf",
        {"sudo", "git", "clone", "-b", "master", "--depth", "1", "https://github.com/keyitdev/sddm-astronaut-theme.git", "/usr/share/sddm/themes/sddm-astronaut-theme"},
        {"sudo", "cp", "-r", "/usr/share/sddm/themes/sddm-astronaut-theme/Fonts/*", "/usr/share/fonts/"},
        "/usr/share/sddm/themes/sddm-astronaut-theme/metadata.desktop"
    }
}

m.screen_share.relationship=Relationship.ASSOCIATED
m.screen_share={
    xdg_desktop_portal_gtk=aor,
    xdg_desktop_portal_hyprland=aor
}

m.bar={
    waybar={
        Repo.AOR,
        units={"waybar.service", Scope.SINGLE_USER},
        single_user_config={"~/.config/waybar"}
    },
    -- ashell=aur,
    -- hyprpanel=aor,
}

m.widget_system={
    -- quickshell=aor,
    -- eww=aor,
    -- ags=aor,
}

m.color_extractor={
    -- pywall=aor,
    -- wallust=aor,
    -- hellwal=aor,
    matugen={Repo.AOR, single_user_config={"~/.config/matugen"}},
    -- python_colorthief=aor,
    -- kde_material_you_colors=aor,
}
m.qt_theme.relationship=Relationship.ASSOCIATED
m.qt_theme={
    qt5ct={Repo.AOR, single_user_config={"~/.config/qt5ct"}},
    qt6ct={Repo.AOR, single_user_config={"~/.config/qt6ct"}},
    hyprqt6engine=aor,
    kvantum={Repo.AOR, single_user_config={"~/.config/Kvantum"}},
}

m.gtk_theme.relationship=Relationship.ASSOCIATED
m.gtk_theme={
    gtk={
        single_user_config={
            "~/.config/gtk-3.0",
            "~/.config/gtk-4.0"
        }
    },
    nwg_look={Repo.AOR, single_user_config={"~/.config/nwg-look"}},
    orchis_theme={
        Repo.AOR,
        supporters={
            vimix_cursors=aor,
            tela_circle_icon_theme=aor,
        },
    },
    gnome_themes_extra=aor,
}

m.wallpaper.resources={"https://wiki.hypr.land/Useful-Utilities/Wallpapers/"}
m.wallpaper.setter={
    -- swww=aor,
    -- swaybg=aor,
    -- awww_bin=aur,
    -- mpvpaper=aur,
    -- wpaperd=aur,
    hyprpaper={
        Repo.AOR,
        units={"hyprpaper.service", Scope.SINGLE_USER},
        single_user_config={"~/.config/hypr/hyprpaper.conf"}
    }
}

m.wallpaper.picker={
    rofi={Repo.AOR, single_user_config={"~/.config/rofi"}},
    -- waypaper=aur,
    -- wallrizz=aur,
    -- waytrogen_bin=aur,
}

m.screen_lock.hyprlock={Repo.AOR, single_user_config={"~/.config/hypr/hyprlock.conf"}}
m.polkit.hyprpolkitagent={Repo.AOR, units={"hyprpolkitagent.service", Scope.SINGLE_USER}}
m.idle.hypridle={
    Repo.AOR,
    units={"hypridle.service", Scope.SINGLE_USER},
    single_user_config={"~/.config/hypr/hypridle.conf"}
}
m.backlight.hyprsunset={
    Repo.AOR,
    units={"hyprsunset.service", Scope.SINGLE_USER},
    single_user_config={"~/.config/hypr/hyprsunset.conf"}
}

m.launcher={
    fuzzel={Repo.AOR, single_user_config={"~/.config/fuzzel"}},
    hyprlauncher={Repo.AOR, single_user_config={"~/.config/hypr/hyprlauncher.conf"}},
}

m.notification={
    -- fnott=aor,
    -- dunst={Repo.AOR, single_user_config={"~/.config/dunst"}},
    -- mako={Repo.AOR, units={"mako.service", Scope.SINGLE_USER}, single_user_config={"~/.config/mako"}},
    swaync={Repo.AOR, units={"swaync.service", Scope.SINGLE_USER}, single_user_config={"~/.config/swaync"}},
}

m.clipboard.copy_paste.wl_clipboard=aor
m.clipboard.history={
    -- copyq=aor,
    -- TODO: how and should i declate it config in hyprland?
    cliphist=aor,
    -- nwg_clipman=aor,
    -- wl_clllip_persist=aor,
}

m.input.method.fcitx5={
    Repo.AOR,
    single_user_config={
        "~/.config/fcitx5/config",
        "~/.config/fcitx5/profile"
    },
    supporters={
        fcitx5_unikey=aor,
        -- fcitx5_lotus_bin=aur, need quite more config here
        fcitx5_configtool=aor
    }
}
m.input.remapper={
    kanata_bin={
        Repo.AUR, single_user_config={"~/.config/kanata"},
        multiple_user_config={
            "/etc/udev/rules.d/90-uinput.rules",
            -- ACTION=="add", KERNEL=="uinput", RUN+="/usr/bin/setfacl -m u:victor:rw /dev/uinput"
        },
        scripts={
            "sudo udevadm control --reload-rules",
            "sudo udevadm trigger",
            "sudo modprobe -r uinput",
            "sudo modprobe uinput"
        },
        groups={
            "input",
            "uinput",
        }
    },
    -- kanata_bin={
    --     Repo.AUR, single_user_config={"~/.config/kanata"},
    --     multiple_user_config={
    --         "/etc/udev/rules.d/99-uinput.rules"
    --         -- KERNEL=="uinput", MODE="0660", GROUP="uinput", OPTIONS+="static_node=uinput"
    --     },
    --     scripts={
    --         "sudo udevadm control --reload",
    --         "sudo udevadm trigger --verbose --sysname-match=uinput",
    --
    --         "sudo modprobe uinput"
    --     },
    --     groups={
    --         "input",
    --         "uinput",
    --     }
    -- },
    -- wlr_which_key=aur,
    -- xremap_hypr_bin=aur,
    -- xremap_wlroots_bin=aur,
    -- input_remapper_bin=aur,
    -- keyd=aor,
    -- kmonad=aor,
}

m.keylogger={
    logkeys=aur,
    whatpulse=aur,
    osa=aur,
    keymouse_logger={repo=Repo.GTIHUB}
}

m.font.noto={
    noto_fonts=aor,
    noto_fonts_cjk=aor,
    noto_fonts_emoji=aor,
}
m.font.ttf={
    ttf_opensans=aor,
    ttf_fira_code=aor,
    ttf_cascadia_code_nerd=aor,
    ttf_jetbrains_mono_nerd=aor,
}
m.font.tex.tex_gyre_fonts=aor

m.shell={
    zsh=aor,
    fish=aor,
    bash={Repo.AOR, single_user_config={"~/.bashrc", "~/.bash_profile"}},
    nushell={
        Repo.AOR,
        single_user_config={
            "~/.config/nushell/env.nu",
            "~/.config/nushell/config.nu",
        },
        personalized_data={
            "~/.config/nushell/history.txt",
            "~/.config/nushell/history.sqlite3"
        }
    }
}

m.shell.history.atuin={
    Repo.AOR,
    supporters={
        nushell={
            Repo.AOR,
            single_user_config={
                "~/.config/nushell/config.nu",
            },
            personalized_data={
                "~/.config/nushell/history.txt",
                "~/.config/nushell/history.sqlite3"
            }
        }
    },
    scripts={
        "mkdir ~/.local/share/atuin/",
        "atuin init nu | save ~/.local/share/atuin/init.nu"
    }
}

m.shell.prompt.starship={
    Repo.AOR,
    single_user_config={"~/.config/starship.toml"},
    scripts={
        -- "mkdir ($nu.data-dir | path join "vendor/autoload")",
        -- "starship init nu | save -f ($nu.data-dir | path join "vendor/autoload/starship.nu")"
    },
    supporters={
        nushell={
            Repo.AOR,
            single_user_config={"~/.config/nushell/config.nu"},
        }
    },

}

m.terminal.emulator={
    foot={
        Repo.AOR,
        units={"foot-server.service", Scope.SINGLE_USER},
        single_user_config={"~/.config/foot"}
    },
    wezterm_git={
        Repo.AUR,
        single_user_config={"~/.config/wezterm"}
    },
    kitty={Repo.AOR, single_user_config={"~/.config/kitty"}},
    aclacritty=aor,
    ghostty={Repo.AOR, units={"app-com.mitchellh.ghostty.service", Scope.SINGLE_USER}},
}

m.terminal.multiplexer={
    tmux={Repo.AOR, single_user_config={"~/.tmux.conf"}},
    zellij=aor,
}

m.file.manager={
    thunar={
        Repo.AOR,
        supporters={
            thunar_volman=aor,
            thunar_archive_plugin=aor,
            thunar_media_tags_plugin=aor,
            -- catfish=aor,
            -- plocate=aor,
            -- zeitgeist=aor,
        }
    },
    yazi={
        Repo.AOR,
        single_user_config={"~/.config/yazi"},
        supporters={
            ["7zip"]=aor,
            chafa=aor,
            ffmpeg=aor ,
            jq=aor,
            poppler=aor,
            resvg=aor,
            imagemagick={Repo.AOR, supporters={libjpeg_turbo=aor}}
        }
    },
    superfile=aor
}
m.file.misc={
    xdg_user_dirs={Repo.AOR, units={"xdg-user-dirs-update.service", Scope.SINGLE_USER}},
    czkawka_gui_bin=aur,
}

m.file.thumnail.tumbler={
    Repo.AOR,
    units={"tumblerd.service", Scope.SINGLE_USER},
    supporters={
        ffmpegthumbnailer=aor,
        freetype2=aor,
        libgepub=aor,
        libgsf={Repo.AOR},
        libopenraw=aor,
        poppler_glib=aor,
        libarchive=aor,
        -- ueberzugpp=aor,
    }
}

m.media.videp={
    mpv={Repo.AOR, single_user_config={"~/.config/mpv"}},
    -- vlc=aor,
    -- clapper=aor,
}
m.media.image={
    mpv={Repo.AOR, single_user_config={"~/.config/mpv"}},
    -- imv=aor,
    -- feh=aor,
    -- gthumb=aor,
    -- swayimg=aor,
}

m.screen.capture={
    -- grim=aor,
    -- flameshot={Repo.AOR},
    hyprshot={Repo.AOR, supporters={hyprpicker=aor}}
}
m.screen.crop.slurp=aor
m.screen.annotate={
    swappy=aor,
    satty=aor,
}

m.screen.record={
    -- wl_screenrec=aur,
    -- obs_studio=aor,
    -- wf_recorder=aor,
    gpu_screen_recorder={Repo.AOR, supporters={gpu_screen_recorder_ui=aor}}
}

m.browser.gui={
    firefox={Repo.AOR, supporters={speech_dispatcher=aor}},
    brave_bin=aur,
    zen_browser_bin=aur,
    google_chrome_bin={Repo.AUR, single_user_config={"~/.config/chrome-flags.conf"}},
    helium_browser_bin={Repo.AUR, single_user_config={"~/.config/helium-flags.conf"}},
    thorium_browser_bin={Repo.AUR, single_user_config={"~/.config/thorium-flags.conf"}},
    microsoft_edge_stable_bin=aur,
}

m.misc={
    -- hugo=aor,
    anki=aor,
    okular=aor,
    mediawriter=aor,
    cava={Repo.AOR, single_user_config={"~/.config/cava"}},
    electron={Repo.AOR, single_user_config={"~/.config/electron-flags.conf"}},
    --TODO: this recipes need extension too, how to write
    python_pywalfox=aur
}
